dm/pkg/schema/tracker.go

```diff
@@ -399,19 +400,28 @@ func IsTableNotExists(err error) bool {
 func (tr *Tracker) Reset() error {
 	tr.se.SetValue(sessionctx.QueryString, "skip")
 	allDBs := tr.dom.InfoSchema().AllSchemaNames()
-	ddl := tr.dom.DDL()
 	for _, db := range allDBs {
 		dbName := model.NewCIStr(db)
 		if filter.IsSystemSchema(dbName.L) {
 			continue
 		}
-		if err := ddl.DropSchema(tr.se, dbName); err != nil {
+		if err := dropDatabase(tr.dom, tr.se, dbName.L); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+func dropDatabase(dom *domain.Domain, se session.Session, db string) error {
+	//
+	query := fmt.Sprintf("DROP DATABASE IF EXISTS %s", dbutil.ColumnName(db))
+	stmt, err := stmtFromQuery(query)
+	if err != nil {
+		return err
+	}
+	return dom.DDL().DropSchema(se, stmt.(*ast.DropDatabaseStmt))
```

Comment:
we can manually build an ast.DropDatabaseStmt literal

ast.DropDatabaseStmt{Name: model.NewCIStr(db), IfExists: true}

Reply:
in that way, we're depending on `DropSchema` doesn't use other fields in `stmt`(although not used for now)

Reply:
there's no other fields in DROP DATABASE, https://dev.mysql.com/doc/refman/8.0/en/drop-database.html